common/src/main/scala/explore/common/UserPreferencesQueries.scala

```diff
@@ -422,12 +450,26 @@ object UserPreferencesQueries:
         w.width.assign
       )
 
-  extension (w: TableColumnPref)
-    def toInput(u: User.Id, t: TableId): LucumaTableColumnPreferencesInsertInput =
-      LucumaTableColumnPreferencesInsertInput(
-        userId = u.show.assign,
-        tableId = t.assign,
-        columnId = w.columnId.value.assign,
-        visible = w.visible.assign,
-        sorting = w.sorting.orIgnore
+  extension (tableColsPrefs: List[TableColumnPreferencesQuery.Data.LucumaTableColumnPreferences])
```

Comment:
Is this more of a UI syntax rather than queries?

Reply:
Not sure, these are extensions to the result type of `TableColumnPreferencesQuery`.

Reply:
yes but mixed with UI code for the table

Reply:
Yup, it translates the query result into table state. I'm open to moving it elsewhere. Do you have a suggestion? Its own file maybe?

Reply:
I started extracting this and realized the whole implementation of `TableStateStore` (lines 392-445) has this characteristic of addressing both the user preferences DB and the table state. I suggest we leave everything together, either here, or moving the whole `TableStateStore` + extensions into a `TableColumnUserPreferences`.

After all, interacting with table state is as much UI as interacting with the grid layout or charts.

Maybe it will look better if we evolve the table facade to use friendlier types other than the `raw` ones, which I'd like to do at some point.

Reply:
👍